Lock specific versions for Docker tools and image size improvements

anton-kasperovich opened the pull request:
- Fixed #3 , removed broken url as default repo now contains "latest" Git version, so it's not needed anymore
- Specified and locked version for (previously every new build might download newer version of Docker or AWS CLI for example): AWS CLI 1.10.19, Docker 1.10.3, Docker Machine 0.6.0, Docker Compose 1.6.0
- Added "yum clean all" to reduce image size

Commit:
Added specific versions for Docker tools and image size improvements

anton-kasperovich commented:
Forgot to mention that from Docker image perspective you wouldn't find any difference, all software inside a container would be absolutely the same as it was previously, more technical improvement rather than some features. My point is that it's minor release.

zrks commented:
Looks good - I also made similar improvements locally; bringing version as variables is always good

anton-kasperovich commented:
@zrks please, don't be shy to send a pull request next time, i'm happy to help you if it's needed

anton-kasperovich commented:
Updated this PR, moved Docker ENV variables for better cache layer support if we would need to change only version of Docker. Thanks @nickdgriffin for the advice.

nickdgriffin commented:
LGTM.
